src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala

```diff
@@ -84,10 +93,34 @@ class Submitter(stagingBaseDir: File,
     _ <- isValid(stageBag)
     // EASY-1464 3.3.7 checksums
     _ <- samePayloadManifestEntries(stageBag, draftBag)
+    _ <- stageBag.baseDir.parent.walk().toStream.map(setRights).find(_.isFailure).getOrElse(Success(()))
```

Comment:
Are you 100% sure that **in all circumstances** the `Iterator` produced by `.walk()` is closed? I currently don't have the mental capacity to proof this to myself 😞 
Concretely: if this fails (you chose 'fail-fast'), is the underlying connection to the file system closed?

Reply:
_Thought from later in this review:_

Perhaps I'm a bit conservative here, but could you take another look at [the code in `easy-split-multi-deposit`](https://github.com/DANS-KNAW/easy-split-multi-deposit/blob/73189001217c2bf31b487eb8356f76ea4e9ffc31/src/main/scala/nl.knaw.dans.easy.multideposit/actions/SetDepositPermissions.scala#L72-L90) and the context in which this is used. As you see we use a `SimpleFileVisitor` from `java.nio.file` here in combination with `Files.walkFileTree`. I'm wondering if this wouldn't be safer to use instead, especially in light of the resource closing mentioned in my previous comment.

Reply:
Under the hood `walk` uses a java stream. From [stack-overflow](https://stackoverflow.com/questions/38339440/how-to-convert-a-java-stream-to-a-scala-stream): "both ways will fully consume the Java Stream".

Not yet sure this applies also to my `tostream`, which by the way is not required to compile. If with toStream the underlying java stream is also completely consumed, the fail-fast approach only prevents further calls to `setRights` after a failure.

Reply:
ps: what's under the hood
https://github.com/pathikrit/better-files/blob/4f2d9b5adf12753af6b1487bfed87e8be1d30073/core/src/main/scala/better/files/File.scala#L783-L784
The Iterator is actually a java.util.stream.Stream[Path]

Reply:
almost... 😃 what you don't see on GitHub is that there is an implicit conversion going on here. Open the same file in IntelliJ and turn on the implicit conversions (https://www.jetbrains.com/help/idea/edit-scala-code.html#scala_hints)